src/internal/rest.rs

```diff
@@ -135,98 +399,156 @@ impl<'a> IronCoreRequest<'a> {
-        builder = builder.body(body.to_vec());
-
-        let req = builder
-            .headers(RAW_BYTES_HEADERS.clone())
-            .headers(auth.to_header());
-        IronCoreRequest::send_req(req, error_code.clone(), move |server_resp| {
-            IronCoreRequest::deserialize_body(server_resp, error_code.clone())
-        })
+            Some(body),
+            None,
+            error_code,
+            auth_b,
+            move |server_resp| IronCoreRequest::deserialize_body(server_resp, error_code),
+        )
+    }
+    pub fn post_raw<B: DeserializeOwned + 'a>(
+        &self,
+        relative_url: &str,
+        body: &'a [u8],
+        error_code: RequestErrorCode,
+        auth_b: AuthV2Builder<'a>,
+    ) -> impl Future<Item = B, Error = IronOxideErr> + 'a {
+        use publicsuffix::IntoUrl;
+
+        let make_req = || {
+            Ok({
+                // build up a request...
+                let mut req = ARequest::new(
+                    Method::POST,
+                    format!("{}{}", self.base_url(), relative_url)
+                        .into_url()
+                        .map_err(|e| IronOxideErr::from((e, error_code)))?,
+                );
+                *req.body_mut() = Some(body.to_vec().into());
+                (req, body.to_vec())
+            })
+        };
+        make_req()
+            .into_future()
+            .and_then(move |(mut req, body_bytes)| {
+                // use the completed request to finish authorization v2 headers
+                let maybe_auth = SignatureUrlString::new(req.url().as_str())
+                    .map(|sig_url| {
+                        auth_b.finish_with(sig_url, req.method().clone(), Some(&body_bytes))
+                    })
+                    .map_err(|e| IronOxideErr::from((e, error_code)));
+
+                // we only support Authorization::Version2 with this call
+                match maybe_auth {
+                    Ok(auth) => {
+                        if let Authorization::Version2 {
+                            user_context,
+                            request_sig,
+                        } = &auth
+                        {
+                            match user_context.to_header(error_code) {
+                                Ok(user_context_header) => {
+                                    replace_headers(req.headers_mut(), user_context_header);
+                                    replace_headers(req.headers_mut(), DEFAULT_HEADERS.clone());
+                                    replace_headers(req.headers_mut(), auth.to_auth_header());
+                                    replace_headers(req.headers_mut(), request_sig.to_header());
+
+                                    Either::A(Self::send_req(req, error_code, move |server_resp| {
+                                        IronCoreRequest::deserialize_body(
+                                            server_resp,
+                                            error_code.clone(),
```

Comment:
Why does this clone the error code when the previous method doesn't?

Reply:
It doesn't need to because that enum is also `Copy`. I'm not sure why it's also `Clone` to be honest. Either way the effect is the same. Removed the `clone()`